File: functions_matlab/draw_tc_samples.m
function draw_tc_samples(simulated_activity_rest, cortex_ind, name, T, num_samples, is_cropped)
    % draw_tc_samples  Plot time‐courses of evenly spaced cortex vertices
    %
    %   draw_tc_samples(simulated_activity_rest, cortex_ind, name, T, num_samples)
    %
    %   Inputs:
    %     simulated_activity_rest  [N×M] matrix of N vertices × M time‐points
    %     cortex_ind               vector of valid vertex indices (subset of 1…N)
    %     name                     string used to name the output file
    %     T                        vector of time‐points (length M)
    %     num_samples              number of vertices to sample & plot
    %
    %   Example:
    %     draw_tc_samples(activity, cortex_ind, 'run1', time_vector, 8);
    % disp(T);
    if nargin < 5 || isempty(num_samples)
        num_samples = 6;   % default to 6 samples
    end

    % ensure cortex_ind is sorted and within valid range
    cortex_ind = cortex_ind(:);
    cortex_ind = cortex_ind(cortex_ind >= 1 & cortex_ind <= size(simulated_activity_rest,1));

    % how many cortex vertices we have
    Nc = numel(cortex_ind);
    if num_samples > Nc
        warning('Requested %d samples but only %d cortex vertices available. Reducing to %d.', ...
                num_samples, Nc, Nc);
        num_samples = Nc;
    end

    % pick indices evenly spaced within cortex_ind
    sample_positions = round(linspace(1, 20000, num_samples));
    idx_samples = cortex_ind(sample_positions);

    % determine subplot grid (near square)
    n = numel(idx_samples);
    n_cols = 2;
    n_rows = ceil(n / n_cols);

    % create figure
    fig_raw = figure('Name', name, ...
                     'NumberTitle', 'off');

    if is_cropped
        T = T(:, 300:end);
        simulated_activity_rest = simulated_activity_rest(:, 300:end);
        name = sprintf("%s_cropped", name);
    end
    % plot each sampled cortex vertex
    for k = 1:n
        vi = idx_samples(k);
        subplot(n_rows, n_cols, k);
        plot(T, simulated_activity_rest(vi, :), 'LineWidth', 1);
        xlim([T(1), T(end)]);
        xlabel('Time (s)');
        ylabel('Activity');
        title(sprintf('Vertex %d', vi));
        grid on;
    end

    % save to file (as PNG)
    saveas(fig_raw, fullfile('outputs', sprintf('%s.png', name)));
end

File: functions_python/draw_tc_samples.py
import numpy as np
import matplotlib.pyplot as plt
import os

def draw_tc_samples(simulated_activity_rest, cortex_ind, name, T, num_samples=6, is_cropped=False):
    cortex_ind = np.array(cortex_ind).flatten()
    cortex_ind = cortex_ind[(cortex_ind >= 1) & (cortex_ind <= simulated_activity_rest.shape[0])]

    Nc = len(cortex_ind)
    if num_samples > Nc:
        print(f'Warning: Requested {num_samples} samples but only {Nc} cortex vertices available. Reducing to {Nc}.')
        num_samples = Nc

    sample_positions = np.round(np.linspace(1, min(20000, Nc), num_samples)).astype(int) - 1
    idx_samples = cortex_ind[sample_positions]

    n = len(idx_samples)
    n_cols = 2
    n_rows = int(np.ceil(n / n_cols))

    if is_cropped:
        T = T[300:]
        simulated_activity_rest = simulated_activity_rest[:, 300:]
        name = f"{name}_cropped"

    fig_raw = plt.figure(figsize=(12, 4*n_rows))
    fig_raw.suptitle(name, fontsize=16)

    for k in range(n):
        vi = idx_samples[k]
        plt.subplot(n_rows, n_cols, k + 1)
        plt.plot(T, simulated_activity_rest[vi-1, :], linewidth=1)
        plt.xlim([T[0], T[-1]])
        plt.xlabel('Time (s)')
        plt.ylabel('Activity')
        plt.title(f'Vertex {vi}')
        plt.grid(True)

    plt.tight_layout()
    os.makedirs('outputs', exist_ok=True)
    plt.savefig(os.path.join('outputs', f'{name}.png'), dpi=300, bbox_inches='tight')
    plt.close()
